base_user_role/models/user.py

```diff
@@ -55,6 +55,11 @@ def set_groups_from_roles(self, force=False):
         If no role is defined on the user, its groups are let untouched unless
         the `force` parameter is `True`.
         """
+        role_groups = {}
```

Comment:
Can you put a comment describing why you need to do that? For further reviews/migrations!

Reply:
We obtain all the groups associated to each role first, so that it is faster to compare later with each user's groups.

Reply:
I am adding this as a comment